### ISSUE TITLE
Converter cannot work in the "giot.switch.v83ksm" well

### ISSUE THREAD

**Issue by DragonsCgithub** [AlexxIT/XiaomiGateway3#1478]
I have been using Gateway3 integration all the time, and it has always worked very well. 

However, recently I added a new wall switch, and the new switch is not supported in Gateway3 integration. 

[giot.switch.v83ksm](https://home.miot-spec.com/spec/giot.switch.v83ksm)

After I tried to read the wiki and create a new `xiaomi_gateway3.py` file in the same-level directory as `configuration.yaml` to write my own `Converter`, but no matter how many times I restarted my Home Assistant, nothing changed. 
<img width="1920" alt="image" src="https://github.com/user-attachments/assets/b552fe4f-7fe4-49e7-b5f1-8845da4b4e31">

There is still only one `Mesh` entity for the switch in the integration. 
<img width="1920" alt="iShot_2024-10-28_15 43 45" src="https://github.com/user-attachments/assets/cb7ca139-9ddc-41c5-9691-28fd0ea26051">

Have I missed something? Thanks for your reply.

The following is my `xiaomi_gateway3.py`:
```
from custom_components.xiaomi_gateway3.core.devices import *

DEVICES = [{
    20448: ["GranwinIoT", "Mesh Smart Triple Wall Switch", "giot.switch.v83ksm"],
    "spec": [
        # 开关设置
        BaseConv("左键", "switch", mi="2.p.1"),
        BaseConv("中键", "switch", mi="3.p.1"),
        BaseConv("右键", "switch", mi="4.p.1"),
        # 开关模式设置
        MapConv("左键开关模式", "select", mi="2.p.2", map={
            0: "普通开关", 
            1: "无线开关", 
            2: "快速断通", 
            3: "点动开关", 
            4: "普通+无线"
        }),
        MapConv("中键开关模式", "select", mi="3.p.2", map={
            0: "普通开关", 
            1: "无线开关", 
            2: "快速断通", 
            3: "点动开关", 
            4: "普通+无线"
        }),
        MapConv("右键开关模式", "select", mi="4.p.2", map={
            0: "普通开关", 
            1: "无线开关", 
            2: "快速断通", 
            3: "点动开关", 
            4: "普通+无线"
        }),
        # 灯光设置
        MapConv("普通开关模式灯光", "select", mi="16.p.1", map={
            0: "跟随", 
            1: "相反", 
            2: "常灭", 
            3: "常亮"
        }),
        MapConv("非普通开关模式灯光", "select", mi="16.p.2", map={
            0: "跟随", 
            1: "相反", 
            2: "常灭", 
            3: "常亮"
        }),
        BoolConv("背光开关", "switch", mi="22.p.1"),
    ]
}] + DEVICES
```

**Comment by AlexxIT** [comment]
Looks okay. Did you reboot Hass?

**Comment by DragonsCgithub** [comment]
> Looks okay. Did you reboot Hass?

yes sure. I have tried to reload the Gateway3 integration and reboot Home Assistant, but every time I open the device details, I am always disappointed.😔

**Comment by AlexxIT** [comment]
You can add your changes to target file directly. You importing it in your code.

**Comment by DragonsCgithub** [comment]
> You can add your changes to target file directly. You importing it in your code.

I am sorry. What is the "target file"?

**Comment by DragonsCgithub** [comment]
I noticed the 'import' at the beginning of the file. Do I need to install relevant Python libraries to implement Converter?

**Comment by DragonsCgithub** [comment]
I tried to use Converter to customize the switch [giot.switch.v52ksm](https://home.miot-spec.com/s/giot.switch.v52ksm) in my hand. 

As a result, it still doesn't work. 

It seems that the `xiaomi_gateway3.py` file is not executed by Home Assistant?

**Comment by AlexxIT** [comment]
This file https://github.com/AlexxIT/XiaomiGateway3/blob/master/custom_components/xiaomi_gateway3/core/devices.py

`xiaomi_gateway3.py` should be executed. You can check logs for any errors about this.

**Comment by DragonsCgithub** [comment]
> This file https://github.com/AlexxIT/XiaomiGateway3/blob/master/custom_components/xiaomi_gateway3/core/devices.py
> 
> `xiaomi_gateway3.py` should be executed. You can check logs for any errors about this.

I just tried adding a custom Converter in `/homeassistant/custom_components/xiaomi_gateway3/core/converters/devices.py`. 

After rebooting, my switch finally displays normally. 🥹

I don't know if the file you mentioned refers to modifying this local file.

 (Sorry, I am a Github novice and I am still learning to use Github.)

**Comment by DragonsCgithub** [comment]
<img width="1920" alt="image" src="https://github.com/user-attachments/assets/d8ab2b76-73c7-407e-86e6-b9ff844a3551">

**Comment by DragonsCgithub** [comment]
Oh! I have found the root of all problems. 

**My integration version is 3.3.3** (_I don't know why the integration has not been checked for updates_). 

However, I used the new version method of `from custom_components.xiaomi_gateway3.core.devices import *` to write the Converter.

That's why my `xiaomi_gateway3.py` didn't work. 

Now I have reinstalled to the latest version of the integration and written `xiaomi_gateway3.py` as above. Everything works as expected!

Thank you, AlexxIT! 😀
